infobserve/processors/yara_processor.py

```diff
@@ -53,45 +50,24 @@ async def process(self):
         The consumer function for the source queue.
         Removes Sources from the Processing Queue, tries to match them
         against the provided Yara rules. Places any matches into the Database
-        Queue for further processing and storage (TBI - Right now, it simply
-        logs matches)
+        Queue(_db_queue)
         """
         APP_LOGGER.info("Processing started. (Using Yara Engine)")
         self._processing = True
 
         # A (practically) infinite amount of items will be processed
-        # unless a STOP command is received, in which case, any remaining items will be
-        # processed and then processing will stop
         items_remaining = sys.maxsize
         items_processed = 0
         while items_processed < items_remaining:
-            remaining = [self._source_queue.get_event(), self._cmd_queue.get_event()]
-            completed_tasks, _ = await asyncio.wait(remaining, return_when=asyncio.FIRST_COMPLETED)
-            for completed_task in completed_tasks:
-                event = await completed_task
-
-                if isinstance(event, YaraProcessor._Command):
-                    if event == YaraProcessor._Command.RECOMPILE:
-                        APP_LOGGER.info("Recompile command received")
-                        self._processing = False
-                        await self.compile_rules()
-                        self._processing = True
-                    elif event == YaraProcessor._Command.STOP:
-                        # When a STOP command is received, all remaining items in the queue will be processed
-                        # and then processing will stop
-                        items_remaining = self._source_queue.events_left()
-                        APP_LOGGER.info(
-                            "Stop command received. Will stop %s", "immediately" if items_remaining == 0 else
-                            (f"after processing {items_remaining} items"))
-                    self._cmd_queue.notify()
-                else:
-                    items_processed += 1
-                    matches = self._engine.match(data=event.raw_content)
-
-                    if matches and not self._has_blacklist(matches):
-                        await self._db_queue.queue_event(ProcessedEvent(event, matches))
-
-                    self._source_queue.notify()
+            event = await self._source_queue.get_event()
+
+            items_processed += 1
+            matches = self._engine.match(data=event.raw_content)
+
+            if matches and not self._has_blacklist(matches):
```

Comment:
Hmm, that's not what I meant when I said to remove the cmd queue :sweat_smile: (and I really don't blame you for not understanding).

My first thought would be to have 2 queues, one for events and one for commands (recompile, stop, etc). This would have worked if asyncio queues had a non-retarded polling method. Because, however, they do, this didn't work. My next thought, because python, was to simply check the class of each object popped from the queue, which worked. Now, we don't really use commands (given how we never gracefully stop execution and all), so we could do without commands in general (which makes the code so much prettier!).
All in all, keep it this way, I just thought I'd clarify what I meant :smile:

Reply:
Yeah asyncio is fucked up :/